lib/libzfs/libzfs_changelist.c

```diff
@@ -409,11 +409,9 @@ change_one(zfs_handle_t *zhp, void *data)
 	 * order to do the rename.  Also, if this is a volume and we're doing
 	 * a rename, then always add it to the changelist.
 	 */
-
-	if (!(ZFS_IS_VOLUME(zhp) && clp->cl_realprop == ZFS_PROP_NAME) &&
-	    zfs_prop_get(zhp, clp->cl_prop, property,
-	    sizeof (property), &sourcetype, where, sizeof (where),
-	    B_FALSE) != 0) {
+	if (zfs_prop_get(zhp, clp->cl_prop, property, sizeof (property),
+	    &sourcetype, where, sizeof (where), B_FALSE) != 0 &&
+	    !(ZFS_IS_VOLUME(zhp) && clp->cl_realprop == ZFS_PROP_NAME)) {
```

Comment:
Rather than reordering these how about initializing `sourcetype` and `share_sourcetype` to `ZPROP_SRC_NONE`.  This also handles the common gcc warning about passing an uninitialized variable to a function as an argument which is expected to set it, and it means less drift from upstream.